### PR TITLE
[cdc] Added postgresql table cdc synchronization action

### PR DESCRIPTION
**Pull request by zhuangchong**
### Purpose

Added postgresql table cdc synchronization action

### CODE REVIEW

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/postgres/PostgresSyncTableActionFactory.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        if (params.has(METADATA_COLUMN)) {
+            action.withMetadataColumns(new ArrayList<>(params.getMultiParameter(METADATA_COLUMN)));
+        }
+
+        if (params.has(TYPE_MAPPING)) {
+            String[] options = params.get(TYPE_MAPPING).split(",");
+            action.withTypeMapping(TypeMapping.parse(options));
+        }
+
+        return Optional.of(action);
```
Comment (TaoZex) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/postgres/PostgresSyncTableActionFactory.java:85]: How about wrapping this part, which is common to sync table action.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/postgres/PostgresActionUtils.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+                    postgresConfig.get(PostgresSourceOptions.SCAN_STARTUP_SPECIFIC_OFFSET_FILE);
+            Integer pos =
+                    postgresConfig.get(PostgresSourceOptions.SCAN_STARTUP_SPECIFIC_OFFSET_POS);
+            sourceBuilder.startupOptions(StartupOptions.specificOffset(file, pos));
+        } else if ("timestamp".equalsIgnoreCase(startupMode)) {
+            sourceBuilder.startupOptions(
+                    StartupOptions.timestamp(
+                            postgresConfig.get(
+                                    PostgresSourceOptions.SCAN_STARTUP_TIMESTAMP_MILLIS)));
+        }
```
Comment (TaoZex) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/postgres/PostgresActionUtils.java:176]: See ：https://github.com/ververica/flink-cdc-connectors/blob/e3d6c7e0aa9b9b8e573311b879f8854d7c88f88f/flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/main/java/com/ververica/cdc/connectors/postgres/table/PostgreSQLTableFactory.java#L207

1705591868234_3D7D86A9-CCCA-435c-BE61-2B716751BA11

Some start modes are not implemented in flink cdc, please fix it.

Comment (zhuangchong) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/postgres/PostgresActionUtils.java]: Thanks @TaoZex, done.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/postgres/PostgresSyncTableActionFactory.java (hunk 1 above)
Comment (zhuangchong) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/postgres/PostgresSyncTableActionFactory.java]: done.

**Review by TaoZex — APPROVED** [review]
Look good to me. The abstract design on the CDC metadata is great! Thanks @zhuangchong

File: paimon-flink/paimon-flink-cdc/src/test/resources/postgres/sync_table_setup.sql [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+-- However, in this database we'll grant the test user 'paimonuser' all privileges:
+--
+
+-- ################################################################################
+--  PostgresSyncTableActionITCase
+-- ################################################################################
+
+CREATE DATABASE paimon_sync_table;
+
+\c paimon_sync_table;
```
[the remark this answers is not in the document]
Comment (zhuangchong) [reply paimon-flink/paimon-flink-cdc/src/test/resources/postgres/sync_table_setup.sql:28]: It is the script syntax for PG to switch databases. It is used to switch databases and can run normally.

File: paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/mysql/MySqlActionITCaseBase.java [hunk 4]
```diff
@@ -84,6 +84,11 @@ protected Map<String, String> getBasicMySqlConfig() {
         config.put("password", PASSWORD);
         // see mysql/my.cnf in test resources
         config.put("server-time-zone", ZoneId.of("America/New_York").toString());
+
+        // TODO When setting the parameter scan.newly-added-table.enabled=true in version 2.4.2, the
+        // Insert data inserted after the newly created table cannot be captured. When set to false,
+        // the mysql cdc works normally.
+        config.put("scan.newly-added-table.enabled", "false");
```
[the remark this answers is not in the document]
Comment (zhuangchong) [reply paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/mysql/MySqlActionITCaseBase.java:91]: The upgrade of mysql cdc to 2.4.2 is due to the incompatibility between the versions of debezium, the dependency package of pgcdc and mysql cdc. The method for pgcdc to use incremental snapshots is only available in version 2.4.x, so the versions of mysqlcdc and pgcdc are both setup 2.4.2

**Review by yuzelin — APPROVED** [review]
+1. I have tested with mysql-cdc 2.3.0. It's compatible.

**Pull request merged**